tracer/src/Datadog.Trace/Ci/Test.cs

```diff
@@ -0,0 +1,268 @@
+    /// Set Test parameters
+    /// </summary>
+    /// <param name="parameters">TestParameters instance</param>
+    public void SetParameters(TestParameters parameters)
+    {
+        if (parameters is not null)
+        {
+            var tags = (TestSpanTags)_scope.Span.Tags;
+            tags.Parameters = parameters.ToJSON();
+        }
+    }
+
+    /// <summary>
+    /// Close test
+    /// </summary>
+    /// <param name="status">Test status</param>
+    public void Close(TestStatus status)
+    {
+        Close(status, null, null);
+    }
+
+    /// <summary>
+    /// Close test
+    /// </summary>
+    /// <param name="status">Test status</param>
+    /// <param name="duration">Duration of the test suite</param>
+    public void Close(TestStatus status, TimeSpan? duration)
+    {
+        Close(status, duration, null);
+    }
+
+    /// <summary>
+    /// Close test
+    /// </summary>
+    /// <param name="status">Test status</param>
+    /// <param name="duration">Duration of the test suite</param>
+    /// <param name="skipReason">In case </param>
+    public void Close(TestStatus status, TimeSpan? duration, string? skipReason)
+    {
+        if (Interlocked.Exchange(ref _finished, 1) == 1)
+        {
+            CIVisibility.Log.Warning("Test.Close() was already called before.");
+            return;
+        }
+
+        var scope = _scope;
+        var tags = (TestSpanTags)scope.Span.Tags;
+
+        // Calculate duration beforehand
+        duration ??= _scope.Span.Context.TraceContext.ElapsedSince(scope.Span.StartTime);
+
+        // Set coverage
+        if (CIVisibility.Settings.CodeCoverageEnabled == true && Coverage.CoverageReporter.Handler.EndSession() is Coverage.Models.CoveragePayload coveragePayload)
+        {
+            if (scope.Span is { } span)
+            {
+                coveragePayload.TraceId = span.TraceId;
+                coveragePayload.SpanId = span.SpanId;
+            }
+
+            CIVisibility.Log.Debug("Coverage data for TraceId={traceId} and SpanId={spanId} processed.", coveragePayload.TraceId, coveragePayload.SpanId);
+            CIVisibility.Manager?.WriteEvent(coveragePayload);
+        }
+
+        // Set status
+        switch (status)
+        {
+            case TestStatus.Pass:
+                tags.Status = TestTags.StatusPass;
+                break;
+            case TestStatus.Fail:
+                tags.Status = TestTags.StatusFail;
+                Suite.Tags.Status = TestTags.StatusFail;
+                break;
+            case TestStatus.Skip:
+                tags.Status = TestTags.StatusSkip;
+                tags.SkipReason = skipReason;
+                break;
+        }
+
+        // Finish
+        scope.Span.Finish(duration.Value);
+        scope.Dispose();
+
+        Current = null;
+        CIVisibility.Log.Debug("######### Test Closed: {name} ({suite} | {module})", Name, Suite.Name, Suite.Module.Name);
+    }
+
+    internal void ResetStartDate()
```

Comment:
Sorry, but here's another place I'd like to suggest a refactor 😅 
```suggestion
    internal void ResetStartTime()
```

Reply:
done in https://github.com/DataDog/dd-trace-dotnet/commit/1804c381ada9ab0c3c2fcbab2e47173544f57df9